src/cpu/aarch64/matmul/jit_uni_sparse_matmul.cpp

```diff
@@ -0,0 +1,430 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+* Copyright 2024 FUJITSU LIMITED
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <cassert>
+
+#include "common/c_types_map.hpp"
+#include "common/dnnl_thread.hpp"
+#include "common/math_utils.hpp"
+#include "common/nstl.hpp"
+#include "common/type_helpers.hpp"
+#include "common/utils.hpp"
+
+#include "cpu/aarch64/jit_generator.hpp"
+
+#include "cpu/aarch64/matmul/jit_uni_sparse_matmul.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+namespace matmul {
+
+using namespace dnnl::impl::data_type;
+using namespace Xbyak_aarch64;
+
+struct sparse_matmul_kernel_t : public jit_generator {
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(sparse_matmul_kernel_t);
+
+    struct call_params_t {
+        const int32_t *src_indices;
+        const float *src_values, *wei, *dst;
+        size_t block_size;
+        size_t nnz;
+    };
+
+    sparse_matmul_kernel_t(size_t vlen, const matmul_pd_t *pd)
+        : jit_generator(), vlen_(vlen) {
+        simd_w_ = vlen_ / data_type_size();
+        N_ = pd->dst_md()->dims[1];
+        tail_block_size_ = N() % block_size();
+        tail_size_ = tail_block_size() % simd_w();
+    }
+
+    ~sparse_matmul_kernel_t() override = default;
+
+    void operator()(const call_params_t *p) {
+        return jit_generator::operator()(p);
+    }
+
+    size_t simd_w() const { return simd_w_; }
+    size_t vlen() const { return vlen_; }
+    size_t tail_block_size() const { return tail_block_size_; }
+    size_t tail_size() const { return tail_size_; }
+
+    int data_type_size() const { return sizeof(float); }
+    int index_type_size() const { return sizeof(int32_t); }
+
+    int block_size() const { return vlen(); }
+
+    int unroll_factor() const { return 2; }
+
+    int N() const { return N_; }
+
+protected:
+    size_t N_;
+    size_t vlen_;
+    size_t simd_w_;
+    size_t tail_block_size_;
+    size_t tail_size_;
+};
+
+template <cpu_isa_t isa>
+struct jit_uni_sparse_matmul_kernel_t : public sparse_matmul_kernel_t {
```

Comment:
It looks like this implementation doesn't work at all. Does it make sense to drop this implementation for now and promote it separately when it is ready?